File: server/java/src/main/java/com/onlinepayments/example/service/dtos/hostedcheckout/CreateHostedCheckoutRequestDto.java
package com.onlinepayments.example.service.dtos.hostedcheckout;

import com.onlinepayments.example.service.domain.common.Address;
import com.onlinepayments.example.service.domain.common.enums.Currency;
import com.onlinepayments.example.service.domain.common.enums.Language;

import java.math.BigDecimal;

public record CreateHostedCheckoutRequestDto(
        BigDecimal amount,
        Currency currency,
        Language language,
        String redirectUrl,
        Address shippingAddress,
        Address billingAddress
) { }

File: server/nodejs/src/business/dtos/hosted-checkout/create-hosted-checkout-request-dto.ts
import { Currency } from '../../domain/common/enums/currency';
import { Language } from '../../domain/common/enums/language';
import { AddressDto } from '../common/address-dto';

export type CreateHostedCheckoutRequestDto = {
    amount?: number | null;
    currency?: Currency | null;
    language?: Language | null;
    redirectUrl?: string | null;
    billingAddress?: AddressDto | null;
    shippingAddress?: AddressDto | null;
};
